modules/basics/src/main/java/com/opengamma/basics/date/HolidayCalendar.java

```diff
@@ -202,6 +203,30 @@ public default LocalDate previousOrSame(LocalDate date) {
 
   //-------------------------------------------------------------------------
   /**
+   * Finds the next business day within the month, returning the input date if it is a business day,
+   * or the last business day of the month if the next business day is in a different month.
+   * <p>
+   * Given a date, this method returns a business day.
+   * If the input date is a business day, it is returned.
+   * If the next business day is within the same month, it is returned.
+   * Otherwise, the last business day of the month is returned.
```

Comment:
Worth highlighting that this means it can return an date that is earlier than the input